src/pudl/workspace/datastore.py

```diff
@@ -426,10 +458,12 @@ def get_resources(self, dataset, **kwargs):
 
                     # save with a relative path
                     r["path"] = str(local.relative_to(self.local_path(dataset)))
+                    self.logger.debug(f"resource local relative path: {r['path']}")
                     self.save_datapackage_json(dataset, dpkg)
 
-                r["path"] = str(self.local_path(dataset, r["path"]))
-                yield r
+                r_abspath = copy.deepcopy(r)
+                r_abspath["path"] = str(self.local_path(dataset, r["path"]))
+                yield r_abspath
```

Comment:
this feels confusing to me. it seems confusing to write out one json file and then pass around another... could we deal w/ the relative to abs paths within another method or even in the extract step? or somehow make it more explicit that the resource being returned is different?